contributing.md

```diff
@@ -258,6 +258,10 @@ Example: `Fix typos in Foo scaladoc`
         1. **Component**
         1. **Affects Version**. For Bugs, assign at least one version that is known to exhibit the 
         problem or need the change
+        1. **Label**. Not widely used, except for the following:
+             - `correctness`: a data loss or correctness issue
+             - `release-notes`: the change's effects need mention in release notes
```

Comment:
For this one do we have an official way to write the text that would go into the release notes?

Reply:
Probably a list of JIRAs and short summary of the impact of each so that readers can find things that will affect them and read details. See how behavior changes are handled here: https://spark.apache.org/releases/spark-release-2-3-0.html

Reply:
All the JIRAs that change the behavior needs to be mentioned in release notes. Thus, it is marked as release-notes too. We started this since 2.3.0.

Reply:
I think we need either the PR author or the person who committed the change to actually write some release note blurb for the RM to include. It's not trivial, at times, for the RM to go and figure out what should be in the release note.

(Internally in our jira we have a special field for that, not sure whether we can get that in Apache's instance.)

Reply:
I agree with I think we need to make sure the jira title is clear and then have somewhere we put more specific text about the change so a user can easily find it.  If Apache won't let us do this perhaps in the description of the jira we can add our own section that is like ========== RELEASE NOTES ============

Is this only for "Changes of behavior" then?

Reply:
In Spark SQL, we require the PR author to write the behavior changes in the same PR. https://github.com/apache/spark/blame/master/docs/sql-programming-guide.md#L1877

It is very easy for RM to write the release note. How about the other components? I think we can follow the same rule?

Reply:
I don't think that's a good approach. Sometimes the release note label is added after the bug is fixed, and the PR has no information about what should be in the release note (e.g. SPARK-24771).

Sometimes the commit message just says "something should be mentioned in the release notes", instead of actually having something that can be copy & pasted into the release notes. I ran into something like that for 2.3.1 (SPARK-23173). It's much better if the people who are familiar with the change write the actual release note, instead of asking the RM to become familiar with code they may have never seen before and hope they understood what's going on.

It shouldn't be hard to add the note to jira itself. Even if just means you copy & paste a little bit of text from the commit message.

Reply:
We need to record the behavior changes in both the doc and the release notes. It is easier for end users to check the behavior changes. That will be a Single Source of Truth; otherwise, they have to check all the release notes, if they migrate from 2.0 to 2.3. If the release note is out, it is hard for the other to edit the release note. However, we still can change the doc easily. This is what Spark SQL is doing.

Reply:
(This is what I am saying too, or trying to, in the doc. All, feel free to propose better wording.)

Reply:
I think what's missing here is how to provide the information about what's changing. The label just tells us that something is changing, but right now whoever is looking for the release notes (the RM when preparing the release, or some user digging for this stuff) doesn't know where to look for it.

What I'm proposing is that we have a field in jira (or some specially marked comment like Tom suggested) where the description of the changing behavior resides. Whether that's written after the fact or copy & pasted from the commit message, it doesn't matter. I just strongly wish that this information was in the bug itself, where it's much easier to find when coupled with the label.

Reply:
Agree, don't know how to get a new field added; INFRA could help but don't know how easy it is. We can, at least, just say here that this info needs to be in the JIRA or PR?

Reply:
I think it must be in jira, and it can be in the PR (in which case the person closing the bug can copy & paste it).

Reply:
I just checked jira and there's a "Docs Text" field we could re-purpose for this, since it doesn't look like we use it for anything currently.

Reply:
Regarding the behavior changes in Spark SQL, the job of RM is simple. They just need to copy and paste what we added in the sql doc.

Reply:
We shouldn't have different policies for different modules. A release note for core is as important as one for SQL. And right now we don't have a defined policy for anything non-SQL.

If you want to make the SQL approach apply to everything else, then great. Propose that and change the release manager guide, which doesn't mention the SQL stuff at all.